### DISCUSSION TITLE
Good to Batch node features running message passing on a static graph?

### DISCUSSION THREAD

**Discussion by songyesog2000** [pyg-team/pytorch_geometric#4070] [General]
Hi there, I am looking for an example to train GCNN with fixed graph structure while the information on the nodes is dynamic.
I wonder can I batchlize the data with one extra dimension (like general nn batching) and run it on the known fixed graph (message passing process ), rather than transforming multiple graph into a large sparse matrix. Is that reasonable and possible to do so? If there is some example training like this, I would be happy to read those codes.

Essentially, I wondering whether the default batch method (SpMM ish) will reduce the model bias for the graph structure, otherwise, I can just try to batch the data as default.

Thank you so much!

**Comment by rusty1s** [comment]
For some GNNs, it is possible to use `[batch_size, num_nodes, num_features]` as node feature matrix as input, e.g., `GCNConv`, `SAGEConv`, ..., while using a fixed `edge_index` across the batch dimension. Hope this helps :)

**Reply by songyesog2000** [reply]
Will have a look of it thanks!
